src/schemas.ts

```diff
@@ -0,0 +1,71 @@
+import * as yup from "yup";
+import { BigNumber, constants, utils } from "ethers";
+
+const testIsNonZeroAddress = {
+  name: "is-non-zero-address",
+  message: "${path} is not a valid address",
+  test: isNonZeroAddress
+};
+
+const testIsPositiveBigNumberish = {
+  name: "is-big-numberish",
+  message: "${path} is not BigNumberish",
+  test: isPositiveBigNumberish
+};
+
+export const recipientSchema = yup.object({
+  amount: yup.string().test(testIsPositiveBigNumberish).defined(),
+  account: yup.string().test(testIsNonZeroAddress).defined(),
+  metadata: yup
+    .object({
+      amountBreakdown: yup
+        .object({
+          welcomeTravelerRewards: yup
+            .string()
+            .test(testIsPositiveBigNumberish)
+            .default("0"),
+          earlyUserRewards: yup
+            .string()
+            .test(testIsPositiveBigNumberish)
+            .default("0"),
+          liquidityProviderRewards: yup
+            .string()
+            .test(testIsPositiveBigNumberish)
+            .default("0"),
+          communityRewards: yup
+            .string()
+            .test(testIsPositiveBigNumberish)
+            .default("0")
+        })
+        .defined()
+    })
+    .defined()
+});
+
+export const recipientsFileSchema = yup.object({
+  chainId: yup.number().integer().positive().defined(),
+  rewardToken: yup.string().test(testIsNonZeroAddress).defined(),
+  windowIndex: yup.number().integer().min(0).defined(),
+  rewardsToDeposit: yup.string().test(testIsPositiveBigNumberish).defined(),
+  recipients: yup.array().of(recipientSchema).defined()
```

Comment:
I chose yup here because I used it in the past quite often. But I think [zod](https://github.com/colinhacks/zod) is also worth looking into.

Reply:
looked over both yup and zod. i havent really used them, but they look like they do the same thing as [superstruct](https://docs.superstructjs.org/), but with more refinements and different apis.  theres also another worth considering, that uses typescript types directly: [typescriptjson](https://github.com/samchon/typescript-json). 

because i havent used them all i cant really form a strong opinion around making any of these standard though willing to hear arguments for any of them